File: problem10.py
def primes(n):
	if n<=2:
		return []
	sieve=[True]*(n+1)
	for x in range(3,int(n**0.5)+1,2):
		if sieve[x]:
		    for y in range(3,(n//x)+1,2):
		        sieve[(x*y)]=False
	return [2]+[i for i in range(3,n,2) if sieve[i]]

print sum(primes(2*(10**7)))

# 142913828922

File: Problem10.java
public class Problem10 {

	public static void main(String args[]) {

		int border = 2000000;

		// Sieve of Eratosthenes
		boolean[] prime = new boolean[border+1];
		for(int i = 2; i <= border; i++) {
			prime[i] = true;
		}

		for(int i = 2; i*i < border; i++) {
			if(prime[i]) {
				int j = i;
				while(i*j <= border) {
					prime[i*j] = false;
					j++;
				}
			}		
		}

		long sum = 0L; // Overflow if int
		for(int i = 2; i <= border; i++) {
			if(prime[i]) sum += i;
		}

		System.out.println(sum);

	} 

}
